android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/ShellActivity.java

```diff
@@ -53,182 +53,182 @@
-            	final String instanceId = uri.getQueryParameter("instanceid");
-            	
-            	new AsyncTask<Void, Void, Void>() {
-                    protected Void doInBackground(Void... v) {
-                        resolveInstance(instanceId);
-                        resolveRecords();
-                        return null;
-                   }
-                   
-                   @Override
-                   protected void onPostExecute(Void v) {
-                      finish();
-                   }
-                }.execute();
-            }
-            super.onPageFinished(view, url);
-        }
-        
-        private void resolveInstance(String instanceId) {
-    		Connection connection = ConnectionFactory.getUnauthenticatedConnection();
-    		connection.setAppId(HealthVaultApp.getInstance().getSettings().getMasterAppId());
-    		HVInstance instance = new HVInstanceResolver(connection)
-    				.getInstanceForId(instanceId);	
-    		HealthVaultSettings settings = HealthVaultApp.getInstance().getSettings();
-    		settings.setServiceUrl(instance.getPlatformUri().toString());
-    		settings.setShellUrl(instance.getShellUri().toString());
-    		settings.setConnectionStatus(ConnectionStatus.Connected);
-    		settings.save();
-        }
-        
-        private void resolveRecords() {
-    		HealthVaultApp hvApp = HealthVaultApp.getInstance();
-
-        	RequestTemplate requestTemplate = new RequestTemplate(
-    				hvApp.getConnection());
-    		
-    		GetAuthorizedPeopleRequest request = new GetAuthorizedPeopleRequest(
-    				new GetAuthorizedPeopleParameters());
-    		
-    		GetAuthorizedPeopleResponse response = requestTemplate.makeRequest(
-    				request, 
-    				GetAuthorizedPeopleResponse.class);
-    		
-    		hvApp.resolvePersonInfoList();
-        }
-        
-        /* (non-Javadoc)
-         * @see android.webkit.WebViewClient#shouldOverrideUrlLoading(android.webkit.WebView, java.lang.String)
-         */
-        @Override
-        public boolean shouldOverrideUrlLoading(WebView view, String url) {
-            view.loadUrl(url);
-            
-            return true;
-        }
-    }
+	/** The web view. */
+	private WebView webView;
+
+	/**
+	 * Creates the app auth intent.
+	 *
+	 * @param ctx the ctx
+	 * @param appId the app id
+	 *
+	 * @return the intent
+	 */
+	public static Intent createAppAuthIntent(Activity ctx, String appId) {
+		String queryString =
+			HealthVaultApp.getInstance().getSettings().getShellUrl()
+			 + "/redirect.aspx?target=APPAUTH&targetqs="
+			 + "?appid=" + appId;
+
+		Intent intent = new Intent(ctx, ShellActivity.class);
+		intent.putExtra("url", queryString);
+
+		return intent;
+	}
+
+	/**
+	 * Creates the create application intent.
+	 *
+	 * @param ctx the ctx
+	 * @param appToken the app token
+	 * @param appId the app id
+	 *
+	 * @return the intent
+	 */
+	public static Intent createCreateApplicationIntent(
+			Context ctx,
```

Comment:
nit: params can fit on same line, or at least some of them can. Not sure if we have set style saying they need to each be on a newline, but thats more of a c# convention i believe.